### PR TITLE
API provides functionality to create an order

### PR DESCRIPTION
**Pull request by emtalen**
# [PT Story](https://www.pivotaltracker.com/story/show/171467249)
```
As a restaurant API
In order for users to buy food they want from the restaurant
I would like to be able to create an order for a specific user
```
## Changes proposed in this pull request:
* Adds units specs for Orders controller, order model and orderitem model.
* Installed and added order serializer 
* Created controller and updated it with functionality
* Created model for Order and OrderItem
## What I have learned working on this feature:
* How to add association between models and db tables
* How to use a serializer
* How to break down a bigger task on the API side
* How to use the binding.pry in a useful and informative way 
* Got a bigger picture between how to manipulate the backend to match your frontend expectations, for example getting 2 x Dish instead of the dish showing up twice in the order
## Packages/Gems added
* active_model_serializers

### CODE REVIEW

File: app/models/order.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class Order < ApplicationRecord
+  has_many :order_items
+
+  def order_total
+
+  order_items.joins(:product).sum('products.price')
+  end
+end
```
Comment (tochman) [inline app/models/order.rb:7]: Please be mindful of the importance of code formatting

File: config/routes.rb [hunk 2]
```diff
@@ -1,6 +1,10 @@
 Rails.application.routes.draw do
+  namespace :api do
+    get 'orders/create'
+  end
```
Comment (tochman) [inline config/routes.rb]: ?? Why do you add the api namespace twice?

File: spec/requests/api/new_order_spec.rb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    order_id = JSON.parse(response.body)['order']['id']
+    @order = Order.find(order_id)
+  end
+  
+  describe 'POST /api/orders' do
+    it 'responds with success message' do
+      expect(JSON.parse(response.body)['message']).to eq 'This item has been added to your order.'
+    end
+    it 'responds with order id' do
+      expect(JSON.parse(response.body)['order']['id']).to eq @order.id
```
Comment (tochman) [inline spec/requests/api/new_order_spec.rb]: You should probably consider adding a helper to parse the response body to JSON and include it in your `rails_helper`

**Review by tochman — COMMENTED** [review]
Please make sure that your code is properly formatted across the board.

File: config/routes.rb (hunk 2 above)
Comment (emtalen) [reply config/routes.rb]: We have no good answer to that really, but we have corrected it to one time now.

File: app/models/order.rb (hunk 1 above)
Comment (emtalen) [reply app/models/order.rb:7]: Yes, we will.

File: spec/requests/api/new_order_spec.rb (hunk 3 above)
Comment (emtalen) [reply spec/requests/api/new_order_spec.rb]: Thank you for that tip, we have added that now and changed how the spec was written

**Pull request merged**